src/translations/v6/v6.ts

```diff
@@ -0,0 +1,149 @@
+import type { Definition } from '@balena/abstract-sql-compiler';
+import type { ConfigLoader } from '@balena/pinejs';
+
+import {
+	aliasFields,
+	aliasTable,
+	generateAbstractSqlModel,
+	overrideFieldType,
+	renameResourceField,
+} from '../../abstract-sql-utils';
+import * as userHasDirectAccessToApplication from '../../features/applications/models/user__has_direct_access_to__application';
+
+export const toVersion = 'resin';
+
+export const v6AbstractSqlModel = generateAbstractSqlModel(
+	__dirname + '/v6.sbvr',
+);
+
+aliasTable(v6AbstractSqlModel, 'application', 'my application');
+userHasDirectAccessToApplication.addToModel(v6AbstractSqlModel);
+
+overrideFieldType(v6AbstractSqlModel, 'release', 'version', 'JSON');
+
+// Convert concept type fields to integer since that's how they were treated in v6 and below (before ConceptType was correctly treated as a reference field)
+overrideFieldType(v6AbstractSqlModel, 'user', 'actor', 'Integer');
+overrideFieldType(v6AbstractSqlModel, 'device', 'actor', 'Integer');
+overrideFieldType(v6AbstractSqlModel, 'application', 'actor', 'Integer');
+
+for (const resource of ['device', 'application']) {
+	renameResourceField(v6AbstractSqlModel, resource, 'env var name', 'name');
+}
+
+export const v6Translations: ConfigLoader.Model['translations'] = {
+	'my application': {
+		$toResource: `application$${toVersion}`,
+		abstractSql: [
+			'SelectQuery',
+			[
+				'Select',
+				aliasFields(v6AbstractSqlModel, 'application', {
+					'depends on-application': ['Cast', ['Null'], 'Integer'],
+				}),
+			],
+			[
+				'From',
+				['Alias', ['Resource', `application$${toVersion}`], 'application'],
+			],
+			[
+				'Where',
+				[
+					'Exists',
+					[
+						'SelectQuery',
+						['Select', []],
+						[
+							'From',
+							[
+								'Alias',
+								[
+									'Resource',
+									`user-has direct access to-application$${toVersion}`,
+								],
+								'user-has direct access to-application',
+							],
+						],
+						[
+							'Where',
+							[
+								'Equals',
+								[
+									'ReferencedField',
+									'user-has direct access to-application',
+									'has direct access to-application',
+								],
+								['ReferencedField', 'application', 'id'],
+							],
+						],
+					],
+				],
+			],
+		],
+		'depends on-application': ['Cast', ['Null'], 'Integer'],
```

Comment:
Since we already have this in the `aliasFields` above, is it needed here as well?

Reply:
https://github.com/balena-io/open-balena-api/pull/1509